### PR TITLE
Remove DOM dependecy and Refactor codebase

### PR DESCRIPTION
**Pull request by aykutkardas**

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <template>
-  <Tree :nodes="data" />
+  <Tree :nodes="data" :useCheckbox="true" />
```
Comment (alpercun) [inline src/App.vue:2]: I think it would be more correct to write the given `useCheckbox` prop with `kebab-case`.

Comment (umutbozdag) [reply src/App.vue:2]: Actually every prop and event names in the template should change to `kebab-case`. It seems like there is no consistency for now.

Comment (alpercun) [reply src/App.vue:2]: You're right. We can resolve this under a separate issue.

Comment (umutbozdag) [reply src/App.vue:2]: Okay. I think i can handle this.

**Review by umutbozdag — APPROVED** [review]

**Review by alpercun — APPROVED** [review]

**Review by denizcdemirci — APPROVED** [review]

**Pull request merged**